guvnor-asset-mgmt/guvnor-asset-mgmt-client/src/main/java/org/guvnor/asset/management/client/editors/repository/structure/RepositoryStructurePresenter.java

```diff
@@ -187,6 +188,8 @@ public RepositoryStructurePresenter( final RepositoryStructureView view,
         this.workbenchContext = workbenchContext;
         this.wizard = wizard;
 
+        workbenchContext.addChangeHandler( this );
```

Comment:
I think they can be used when branch has been created or a file opened, but when it is possible to just use a change listener it is a lot safer and easier to take that path. Here we are using the context as a central location for the current, well, context. So instead of adding the Observers for each event type that changes the context in each location that follows context changes, this is just a lot safer way.

The problems with the old approach were:
- We had events both notifying about a context change and making a context change. Due to bad naming perhaps, but it was hard to see what event A should be used for. 
- We had events updating the context, even when there was no actual server side changes. Context and what was on the file system became out of sync. Direction of events and change listeners should be server -> context -> anything that cares about context
- Context always knows what is the current state. Previously the classes listening to context listened to the same events as context. Now who knows what picks up the event first and then the screens we have are out of sync

I know you @manstis probably know about most of this. Wrote the above in case anyone alse is listening. I think we should make this a coding pattern for context, kind of like the EventBus that is used for a single editor instance. So instead of going wild with global events we narrow down some areas where event listener or scoped event bus is better.